packages/big-design/src/components/Alert/spec.tsx

```diff
@@ -1,97 +1,109 @@
+  render(<Alert messages={[{ text: 'Warning' }]} type="warning" />);
+
+  const alert = await screen.findByRole('alert');
 
-  expect(container.firstChild).toMatchSnapshot();
-  expect(container.firstChild).toHaveStyle(`border-left: ${theme.spacing.xxSmall} solid ${theme.colors.warning50}`);
+  expect(alert).toMatchSnapshot();
+  expect(alert).toHaveStyle(`border-left: ${theme.spacing.xxSmall} solid ${theme.colors.warning50}`);
 });
 
-test('render info Alert', () => {
-  const { container } = render(<Alert messages={[{ text: 'Info' }]} type="info" />);
+test('render info Alert', async () => {
+  render(<Alert messages={[{ text: 'Info' }]} type="info" />);
 
-  expect(container.firstChild).toMatchSnapshot();
-  expect(container.firstChild).toHaveStyle(`border-left: ${theme.spacing.xxSmall} solid ${theme.colors.primary60}`);
-});
+  const alert = await screen.findByRole('alert');
 
-test('renders with link', () => {
-  const { queryByRole, container } = render(
-    <Alert messages={[{ text: 'Success', link: { text: 'Link', href: '#' } }]} />,
-  );
+  expect(alert).toMatchSnapshot();
+  expect(alert).toHaveStyle(`border-left: ${theme.spacing.xxSmall} solid ${theme.colors.primary60}`);
+});
 
-  expect(container.firstChild).toMatchSnapshot();
+test('renders with link', async () => {
+  render(<Alert messages={[{ text: 'Success', link: { text: 'Link', href: '#' } }]} />);
 
-  const link = queryByRole('link') as HTMLAnchorElement;
+  const alert = await screen.findByRole('alert');
+  const link = await screen.findByRole<HTMLAnchorElement>('link');
 
+  expect(alert).toMatchSnapshot();
   expect(link).toBeInTheDocument();
   expect(link.href).toBe('http://localhost/#');
 });
 
-test('renders with external link', () => {
-  const { queryByRole, container } = render(
+test('renders with external link', async () => {
+  render(
     <Alert messages={[{ text: 'Success', link: { text: 'Link', href: '#', external: true, target: '_blank' } }]} />,
   );
 
-  expect(container.firstChild).toMatchSnapshot();
-
-  const link = queryByRole('link') as HTMLAnchorElement;
+  const alert = await screen.findByRole('alert');
+  const link = screen.getByRole('link') as HTMLAnchorElement;
 
+  expect(alert).toMatchSnapshot();
   expect(link).toBeInTheDocument();
   expect(link.href).toBe('http://localhost/#');
   expect(link.target).toBe('_blank');
   expect(link.querySelector('svg')).not.toBeUndefined();
 });
 
-test('renders header', () => {
-  const { queryByText, container } = render(<Alert header="Header" messages={[{ text: 'Success' }]} />);
+test('renders header', async () => {
+  render(<Alert header="Header" messages={[{ text: 'Success' }]} />);
+
+  const alert = await screen.findByRole('alert');
 
-  expect(container.firstChild).toMatchSnapshot();
-  expect(queryByText('Header')).not.toBeUndefined();
+  expect(alert).toMatchSnapshot();
+  expect(screen.getByRole('heading', { name: /header/i })).not.toBeUndefined();
```

Comment:
Let's assign these to a variable before we expect on them, along with using the `toBeDefined` method.
```tsx
  const alert = await screen.findByRole('alert');
  const heading = screen.getByRole('heading', { name: /header/i });

  expect(alert).toMatchSnapshot();
  expect(heading).toBeDefined();
```